packages/zowe-explorer/src/trees/dataset/DatasetActions.ts

```diff
@@ -1079,6 +1079,109 @@ export class DatasetActions {
+        // Shouldn't happen but just in case
+        if (!selectedText) {
+            Gui.warningMessage(vscode.l10n.t("No selection to open."));
+            return;
+        }
+
+        // If selected text is not a valid data set name or data set(member name)
+        const ds = DatasetUtils.extractDataSetAndMember(selectedText);
+        const hasMember = ds.memberName && ds.memberName.length > 0;
+        if (hasMember) {
+            if (!DatasetUtils.validateMemberName(ds.memberName)) {
+                Gui.warningMessage(vscode.l10n.t("Selection is not a valid data set member name."));
+                return;
+            }
+        } else if (!DatasetUtils.validateDataSetName(ds.dataSetName)) {
+            Gui.warningMessage(vscode.l10n.t("Selection is not a valid data set name."));
+            return;
+        }
+
+        let sessProfileName: string = "";
+        const profiles = Profiles.getInstance();
+        const profileNamesList = ProfileManagement.getRegisteredProfileNameList(Definitions.Trees.MVS);
+        if (doc.uri && FsAbstractUtils.getInfoForUri(doc.uri)?.profileName) {
+            sessProfileName = FsAbstractUtils.getInfoForUri(doc.uri).profileName;
+        }
+
+        // If no profile name or not loaded, prompt user to select one
+        if (!sessProfileName || !profiles.allProfiles.some((p) => p.name === sessProfileName)) {
+            if (profileNamesList.length > 1) {
+                const quickPickOptions: vscode.QuickPickOptions = {
+                    placeHolder: vscode.l10n.t("Select the profile to use to open the data set"),
+                    ignoreFocusOut: true,
+                    canPickMany: false,
+                };
+                sessProfileName = await Gui.showQuickPick(profileNamesList, quickPickOptions);
+                if (!sessProfileName) {
+                    Gui.infoMessage(DatasetActions.localizedStrings.opCancelled);
+                    return;
+                }
+            } else if (profileNamesList.length > 0) {
+                sessProfileName = profileNamesList[0];
+            } else {
+                Gui.showMessage(vscode.l10n.t("No profiles available"));
+                return;
+            }
+        }
+
+        // Get the profile from the session name
+        const sessProfile = profiles.loadNamedProfile(sessProfileName);
+
+        await Profiles.getInstance().checkCurrentProfile(sessProfile);
+        if (Profiles.getInstance().validProfile === Validation.ValidationType.INVALID) {
+            Gui.errorMessage(DatasetActions.localizedStrings.profileInvalid);
+            return;
+        }
+
+        let datasetUri: vscode.Uri;
+        if (hasMember) {
+            datasetUri = vscode.Uri.parse(`${ZoweScheme.DS}:/${sessProfileName}/${ds.dataSetName.toUpperCase()}/${ds.memberName.toUpperCase()}`);
+        } else {
+            datasetUri = vscode.Uri.parse(`${ZoweScheme.DS}:/${sessProfileName}/${ds.dataSetName.toUpperCase()}`);
+        }
```

Comment:
We can replace the use of `let` with `const` here as well as joining variable assignment/declaration:

```suggestion
        const datasetUri = vscode.Uri.parse(hasMember ? `${ZoweScheme.DS}:/${sessProfileName}/${ds.dataSetName.toUpperCase()}/${ds.memberName.toUpperCase()}` : `${ZoweScheme.DS}:/${sessProfileName}/${ds.dataSetName.toUpperCase()}`);
```